### PR TITLE
Implement collateral switch

### PR DESCRIPTION
**Pull request by matzapata**
## Overview

Jira ticket: [SAF-48](https://wakeuplabs.atlassian.net/browse/SAF-48)

Implement collateral switch transaction and add some cleanups to supply hook. I'll replicate for the rest of the hooks and tx assembler. Probably in separate pr to catch different pr as well.

## Demo

https://github.com/user-attachments/assets/acd382dc-7268-4630-890a-799ab3671127

https://github.com/user-attachments/assets/d60d7b8c-6aeb-43b6-aa88-5ec33a104487

### CODE REVIEW

**Review by luchopcerra — APPROVED** [review]
LGTM 🚀

File: apps/frontend/src/utils/aave/AaveSupplyTransactionsFactory.ts [hunk 1]
```diff
@@ -41,14 +43,48 @@ export class AaveSupplyTransactionsFactory {
...
+    if (token.isNative) return this.supplyNative(amount, opts);
+    else return this.supplyToken(token, amount, opts);
+  }
+
+  async collateralSwitch(
+    token: AssetDetailsData,
+    useAsCollateral: boolean,
+    opts?: TransactionFactoryOptions,
+  ): Promise<Transaction[]> {
+    return [
```
Comment (njaume) [inline apps/frontend/src/utils/aave/AaveSupplyTransactionsFactory.ts:57]: Just a question, does It actually return a promise ?

**Review by njaume — COMMENTED** [review]
LGTM! just a comment!

File: apps/frontend/src/utils/aave/AaveSupplyTransactionsFactory.ts (hunk 1 above)
Comment (matzapata) [reply apps/frontend/src/utils/aave/AaveSupplyTransactionsFactory.ts:57]: Well by being `async` technically yes, but I get your point. In this case there's no async work so I can remove that. I think if we keep it is more consistent with the rest of the code and wont harm us, but if you think differently let me know, no problem in removing it )

**Pull request merged**